services/core-web/src/components/Forms/noticeOfWork/EditNoWContacts.js

```diff
@@ -64,7 +64,7 @@ const renderContacts = ({
         {fields
           .map((field, index) => (
             // eslint-disable-next-line react/no-array-index-key
-            <Col lg={12} sm={24} key={index}>
+            <Col lg={12} sm={24} key={`${index}_${fields.length}`}>
```

Comment:
can you revert these changes? This is what caused the party_guids fields to clear when adding a new contact